src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogTests.cs

```diff
@@ -15,62 +16,314 @@ public void EventLogReinitializationException()
-        public void ClearLogTest()
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void ClearLog()
         {
-            if (!AdminHelpers.IsProcessElevated())
-                return;
+            string log = "ClearTest";
+            string source = "Source_" + nameof(ClearLog);
 
-            string source = Guid.NewGuid().ToString("N");
-            string logName = Guid.NewGuid().ToString("N");
             try
             {
-                EventLog.CreateEventSource(source, logName);
-                using (EventLog myLog = new EventLog())
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
                 {
-                    myLog.Source = source;
-                    myLog.WriteEntry("Writing to event log.");
-                    Assert.True(myLog.Entries.Count != 0);
-                    myLog.Clear();
-                    Assert.Equal(0, myLog.Entries.Count);
+                    eventLog.Source = source;
+                    eventLog.Clear();
+                    Assert.Equal(0, eventLog.Entries.Count);
+                    eventLog.WriteEntry("Writing to event log.");
+                    Assert.Equal(eventLog.Entries.Count, 1);
                 }
             }
             finally
             {
                 EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
             }
         }
 
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
         public void ApplicationEventLog_Count()
         {
-            using ( EventLog ael = new EventLog("Application"))
-            { 
-                Assert.InRange(ael.Entries.Count, 1, Int32.MaxValue);
+            using (EventLog eventLog = new EventLog("Application"))
+            {
+                Assert.InRange(eventLog.Entries.Count, 1, Int32.MaxValue);
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void DeleteLog()
+        {
+            string log = "DeleteTest";
+            string source = "Source_" + nameof(DeleteLog);
+
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                Assert.True(EventLog.Exists(log));
+            }
+            finally
+            {
+                EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
+                Assert.False(EventLog.Exists(log));
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void CheckLogName_Get()
+        {
+            using (EventLog eventLog = new EventLog("Application"))
+            {
+                Assert.Equal("Application", eventLog.LogDisplayName);
```

Comment:
This may not pass on a machine that is not the en-US locale. I bet DisplayName is localized. You could try temporarily changing the locale of your windows machine to check all your tests pass.

Reply:
If they don't - one option is to check this only if the CurrentCulture is "en-US"

Reply:
.. and if it isn't, just check it's some non empty string

Reply:
It doesn't fail if the culture is different. it was failing for one api and I corrected that